samcli/lib/sync/watch_manager.py

```diff
@@ -113,17 +113,28 @@ def _add_code_triggers(self) -> None:
             try:
                 trigger = self._trigger_factory.create_trigger(resource_id, self._on_code_change_wrapper(resource_id))
             except (MissingCodeUri, MissingLocalDefinition):
-                LOG.debug("CodeTrigger not created as CodeUri or DefinitionUri is missing for %s.", str(resource_id))
+                LOG.warning(
+                    self._color.yellow("CodeTrigger not created as CodeUri or DefinitionUri is missing for %s."),
+                    str(resource_id),
+                )
                 continue
 
             if not trigger:
                 continue
             self._observer.schedule_handlers(trigger.get_path_handlers())
 
-    def _add_template_trigger(self) -> None:
+    def _add_template_triggers(self) -> None:
         """Create TemplateTrigger and add its handlers to observer"""
-        template_trigger = TemplateTrigger(self._template, lambda _=None: self.queue_infra_sync())
-        self._observer.schedule_handlers(template_trigger.get_path_handlers())
+        stacks = SamLocalStackProvider.get_stacks(self._template)[0]
+        for stack in stacks:
+            template = stack.location
+            template_trigger = TemplateTrigger(template, stack.name, lambda _=None: self.queue_infra_sync())
+            try:
+                template_trigger.raw_validate()
+            except InvalidTemplateFile:
+                LOG.warning(self._color.yellow("Template validation failed for %s in %s"), template, stack.location)
```

Comment:
nit: anything that we can provide a constructive feedback to customer? What they could/should do here?

Reply:
Yep I'll add suggest them to make sure the template path is valid and the template file is in the right format! @mndeveci